### PR TITLE
C library: cleanup and syntax error fixes

### PR DESCRIPTION
**Pull request by tautschnig**
To avoid future syntax errors creeping in, a check is added for Travis.

### CODE REVIEW

File: src/ansi-c/library/getopt.c [hunk 1]
```diff
@@ -12,13 +20,55 @@ inline int getopt(int argc, char * const argv[],
...
+
+/* FUNCTION: getopt_long */
+
+#ifndef __CPROVER_GETOPT_H_INCLUDED
+#include <getopt.h>
+#define __CPROVER_GETOPT_H_INCLUDED
+#endif
+
+inline int getopt_long(int argc, char * const argv[], const char *optstring,
+                const struct option *longopts, int *longindex)
```
Comment (allredj) [inline src/ansi-c/library/getopt.c]: Strange indent

**Comment by allredj** [comment]
Of course the Travis failure needs to be looked at.

**Comment by tautschnig** [comment]
Thanks a lot for the review. I'll await the ongoing merge from master to develop as that will entail some rebase of those pull requests.

File: src/ansi-c/library/getopt.c [hunk 2]
```diff
@@ -12,13 +20,55 @@ inline int getopt(int argc, char * const argv[],
...
+      ++optind;
+    }
+    else
+      optarg=NULL;
+  }
+
+  return result;
+}
+
+/* FUNCTION: getopt_long */
```
Comment (peterschrammel) [inline src/ansi-c/library/getopt.c]: /* */ comments should be avoided

File: src/ansi-c/Makefile [hunk 3]
```diff
@@ -94,14 +94,15 @@ else
...
-		sed -i 's/__sync_/s&/' __libcheck.c ; \
-		sed -i 's/__noop/s&/' __libcheck.c ; \
+		perl -p -i -e 's/(__builtin_[^v])/s$$1/' __libcheck.c ; \
+		perl -p -i -e 's/(__sync_)/s$$1/' __libcheck.c ; \
+		perl -p -i -e 's/(__noop)/s$$1/' __libcheck.c ; \
 		$(CC) -std=gnu99 -E -include library/cprover.h -D__CPROVER_bool=_Bool \
 			-D__CPROVER_thread_local=__thread -DLIBRARY_CHECK -o __libcheck.i __libcheck.c ; \
 		$(CC) -S -Wall -Werror -pedantic -Wextra -std=gnu99 __libcheck.i -o __libcheck.s \
-			-Wno-unused-label -Wno-uninitialized ; \
+			-Wno-unused-label ; \
```
Comment (peterschrammel) [inline src/ansi-c/Makefile]: why did this change?

Comment (tautschnig) [inline src/ansi-c/Makefile]: This warning should no longer be silenced as all local variables are now initialised (with non-deterministic values).

**Pull request merged**